crates/reporting/tests/test_reporting.rs

```diff
@@ -10305,6 +10316,63 @@ All branches in an `if` must have the same type!
             "#
         ),
         @r###"
+    ── TYPE MISMATCH ───────────────────────────────────────── /code/proj/Main.roc ─
+
+    This expression has a type that does not implement the abilities it's expected to:
+
+    7│      myDecoder = decoder
+                        ^^^^^^^
+
+    Roc can't generate an implementation of the `Decode.Decoding` ability
+    for
+
+        { x : A }
+
+    In particular, an implementation for
+
+        A
+
+    cannot be generated.
+
+    Tip: `A` does not implement `Decoding`. Consider adding a custom
+    implementation or `has Decode.Decoding` to the definition of `A`.
+    "###
+    );
+
+    test_report!(
+        record_with_optional_field_types_cannot_derive_decoding,
+        indoc!(
+            r#"
+            app "test" imports [Decode.{Decoder, DecoderFormatting, decoder}] provides [main] to "./platform"
+
+            main =
+                myDecoder : Decoder {x : Str, y ? Str} fmt | fmt has DecoderFormatting
+                myDecoder = decoder
+
+                myDecoder
+            "#
+        ),
+        @r###"
+    ── TYPE MISMATCH ───────────────────────────────────────── /code/proj/Main.roc ─
+
+    This expression has a type that does not implement the abilities it's expected to:
+
+    5│      myDecoder = decoder
+                        ^^^^^^^
+
+    Roc can't generate an implementation of the `Decode.Decoding` ability
+    for
+
+        { x : Str, y ? Str }
+
+    Note: Roc cannot derive decoding for a record with an optional field,
+    which in this case is `.y`. Optional record fields are polymorphic over
+    records that may or may not contain them at compile time, but are not
+    a concept that extends to runtime!
+    That means Roc cannot derive a decoder for a record with an optional
+    value by way of an optional record field. If you want to model the
+    idea that a field may or may not be present at runtime, consider using
+    a `Result`.
```

Comment:
@rtfeldman there was a paper recently that showed that first person "I cannot X" was better received by students than the more impersonal "cannot do X". (https://twitter.com/Felienne/status/1551477360976969728 and https://faculty.washington.edu/ajko/papers/Lee2011Gidget.pdf) What is our policy on voice in error messages?

i personally quite like the elm-style "I cannot X".

Reply:
I find that reasoning compelling! Let's embrace the "I cannot X" style. 👍

Reply:
I'd say go even more casual and use contractions, e.g. "I can't X" instead of "I cannot X"